### PR TITLE
Fix files sync by determining which LMT to use via smb properties flag

### PR DESCRIPTION
**Pull request by adreed-msft**
https://github.com/Azure/azure-storage-azcopy/issues/1956

It turns out having two completely unreliable LMTs doesn't work well.

### CODE REVIEW

**Review by gapra-msft — CHANGES_REQUESTED** [review]
are there any tests we can add to validate this change?

**Comment by vibhansa-msft** [comment]
We shall add test case for both the cases where preferSMBTime  : true/false.

**Comment by tasherif-msft** [comment]
why do we prefer smbLMT over service LMT?

**Comment by adreed-msft** [comment]
> why do we prefer smbLMT over service LMT?

Long and dumb story:
1) We have no control of service LMT
2) We do have control of SMB LMT
3) SMB LMT is what's supposedly "real" if the only tool being used to upload is AzCopy
4) SMB LMT is what's persisted back down to local if SMB properties persistence is enabled

There are two different LMTs-- neither are really truly "accurate". The former only updates when the file is updated on the service & cannot be controlled, and the latter only updates when the client updates it.

**Review by siminsavani-msft — APPROVED** [review]
LGTM! Adding a test case for this scenario would be good!

File: cmd/syncComparator.go [hunk 1]
```diff
@@ -82,16 +83,18 @@ type syncSourceComparator struct {
 	// storing the destination objects
 	destinationIndex *objectIndexer
 
+	preferSMBTime     bool
 	disableComparison bool
 }
 
-func newSyncSourceComparator(i *objectIndexer, copyScheduler objectProcessor, disableComparison bool) *syncSourceComparator {
-	return &syncSourceComparator{destinationIndex: i, copyTransferScheduler: copyScheduler, disableComparison: disableComparison}
+func newSyncSourceComparator(i *objectIndexer, copyScheduler objectProcessor, preferSMBTime, disableComparison bool) *syncSourceComparator {
```
Comment (tasherif-msft) [inline cmd/syncComparator.go:90]: we really should start creating options bags :'(

Comment (adreed-msft) [reply cmd/syncComparator.go:90]: Yes I completely agree.

I would like to take some time in the near future to focus on improving the overall quality of life in the AzCopy codebase, because I generally feel immense pain whenever I have to pass a flag through 8 layers of code and sometimes with 20+ argument functions

File: cmd/syncComparator.go [hunk 2]
```diff
@@ -57,7 +58,7 @@ func (f *syncDestinationComparator) processIfNecessary(destinationObject StoredO
 	// if the destinationObject is present at source and stale, we transfer the up-to-date version from source
 	if present {
 		defer delete(f.sourceIndex.indexMap, destinationObject.relativePath)
-		if f.disableComparison || sourceObjectInMap.isMoreRecentThan(destinationObject) {
+		if f.disableComparison || sourceObjectInMap.isMoreRecentThan(destinationObject, f.preferSMBTime) {
```
Comment (vibhansa-msft) [inline cmd/syncComparator.go:61]: Need test cases for positive and negative scenarios here.

**Review by vibhansa-msft — APPROVED** [review]

**Review by tasherif-msft — APPROVED** [review]
